### PR TITLE
Add tests for SwaggerTypeMapping and SwaggerSchemaFactory

### PR DESCRIPTION
**Pull request by yahehe**
Also slight cleanup on existing tests

### CODE REVIEW

File: test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using Xunit;
+
+namespace Nancy.Swagger.Tests
+{
+    public class SwaggerTypeMappingTests
+    {
+        [Fact]
+        public void ShouldAdd_AndGet_TypeMappings()
+        {
+            SwaggerTypeMapping.ResetMappedTypes();
```
Comment (jnallard) [inline test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs]: Is there a some type of annotation you can use to define a function that should run before (or after) each test?

File: src/Nancy.Swagger/SwaggerTypeMapping.cs [hunk 2]
```diff
@@ -57,5 +57,10 @@ public static Type GetMappedType(Type type, List<Type> previousTypes = null)
 
             return returnType;
         }
+
+        public static void ResetMappedTypes()
+        {
+            TypeMappings.RemoveAll((type) => true);
```
Comment (jnallard) [inline src/Nancy.Swagger/SwaggerTypeMapping.cs]: Can't you just do TypeMappings.Clear()?

File: test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs (hunk 1 above)
Comment (yahehe) [reply test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs]: Not with XUnit unfortunately

File: src/Nancy.Swagger/SwaggerTypeMapping.cs (hunk 2 above)
Comment (yahehe) [reply src/Nancy.Swagger/SwaggerTypeMapping.cs]: Yep, that's better

File: test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs (hunk 1 above)
Comment (yahehe) [reply test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs]: Looks like that was intentional: https://lostechies.com/jimmybogard/2013/09/26/test-styles-and-avoiding-setupteardown/

Comment (yahehe) [reply test/Nancy.Swagger.Tests/SwaggerTypeMappingTests.cs]: Maybe we can migrate to nunit

**Pull request merged**